atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/DualWriteKeyValueService.java

```diff
@@ -104,6 +105,11 @@ public void putUnlessExists(TableReference tableRef, Map<Cell, byte[]> values)
     }
 
     @Override
+    public void checkAndSet(CheckAndSetRequest checkAndSetRequest) {
+        delegate1.checkAndSet(checkAndSetRequest);
```

Comment:
In general, yes, but as the big brother of `putUnlessExists`, `checkAndSet` is special. GitHub lost the comment, but I think there was a previous discussion where we decided to only CAS to one of the two KVSs (because otherwise we can't guarantee atomicity).